dynamiqs/utils/optimal_control.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import torch
+from torch import Tensor
+
+from .._utils import linmap
+from .tensor_types import dtype_complex_to_real, get_cdtype, to_device
+
+__all__ = [
+    'rand_complex',
+    'pwc_pulse',
+]
+
+
+def rand_complex(
+    size: int | tuple[int, ...],
+    *,
+    rmax: float = 1.0,
+    requires_grad: bool = False,
+    seed: int | None = None,
+    dtype: torch.complex64 | torch.complex128 | None = None,
+    device: str | torch.device | None = None,
+) -> Tensor:
+    r"""Returns a tensor filled with random complex numbers uniformly distributed in the
+    complex plane.
+
+    Each element of the returned tensor has a random magnitude between 0 and `rmax` and
+    a random phase. Formally, each element is defined by
+
+    $$
+        x = re^{i\theta}\ \text{with}\
+        \left\{\begin{aligned}
+        r      &= \texttt{rmax} \cdot \sqrt{\texttt{rand(0,1)}} \\
+        \theta &= 2\pi \cdot \texttt{rand(0,1)}
+        \end{aligned}\right.
+    $$
+
+    where $\texttt{rand(0,1)}$ is a random number uniformly distributed between 0 and 1.
+
+    Note:
+        The square root in the definition of the magnitude $r$ ensures that the
+        resulting complex numbers are uniformly distributed in the complex plane.
+
+    Args:
+        size: Size of the returned tensor.
+        rmax: Maximum magnitude.
+        requires_grad: Whether gradients need to be computed with respect to the
+            returned tensor.
+        seed: Seed for the random number generator.
+        dtype: Complex data type of the returned tensor.
+        device: Device of the returned tensor.
```

Comment:
Missing types in paranthesis, like in the rest of the doc.

Reply:
I added the typing to `size` but not the others, we only type complex types in this part of the doc, the rest is in the signature.